### PR TITLE
Add Delete /todos/:id api and delete functionality

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
### Description of Change
- Added Delete api at controller
- Added delete functionality at provider
### Associated Ticket Link:
- https://trello.com/c/UqwfN4Zw

### CODE REVIEW

File: src/tasks/tasks.controller.ts [hunk 1]
```diff
@@ -25,4 +33,9 @@ export class TasksController {
   ): Promise<Task> {
     return this.tasksService.updateTask(+id, updateTaskDto);
   }
+
+  @Delete(':id')
+  async deleteTask(@Param('id') id: string) {
+    return await this.tasksService.deleteTask(+id);
```
Comment (ragib-hossain-sazim) [inline src/tasks/tasks.controller.ts:39]: [JC] what's the use of this `+` operator before `id`

File: src/tasks/tasks.service.ts [hunk 2]
```diff
@@ -25,4 +24,11 @@ export class TasksService {
   async updateTask(id: number, updateTaskDto: UpdateTaskDto): Promise<Task> {
     return await this.taskRepository.updateTask(id, updateTaskDto);
   }
+
+  async deleteTask(id: number) {
+    const task = await this.taskRepository.findOne(id);
```
Comment (ragib-hossain-sazim) [inline src/tasks/tasks.service.ts:29]: `findOne` => `findOneOrFail`

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending comments

File: src/tasks/tasks.service.ts [hunk 3]
```diff
@@ -25,4 +24,11 @@ export class TasksService {
   async updateTask(id: number, updateTaskDto: UpdateTaskDto): Promise<Task> {
     return await this.taskRepository.updateTask(id, updateTaskDto);
   }
+
+  async deleteTask(id: number) {
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.service.ts:28]:
```suggestion
  async deleteOne(id: number) {
```

No need to mention task explicitly in the name as you'd call this method like so: `this.tasksService.deleteOne` in other files.

File: src/tasks/tasks.controller.ts (hunk 1 above)
Comment (faysal-alam-sazim) [reply src/tasks/tasks.controller.ts:39]: Converting id to number

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending merge conflicts

**Pull request merged**